lib/Doctrine/DBAL/Connection.php

```diff
@@ -211,6 +211,15 @@ public function __construct(array $params, Driver $driver, Configuration $config
             unset($this->_params['pdo']);
         }
 
+        if (isset($params["platform"])) {
+            if ( ! $params['platform'] instanceof Platforms\AbstractPlatform) {
+                throw DBALException::invalidPlatformSpecified();
```

Comment:
I've added more detailed exception message if an object is given. If another type is given it fails horrible and the bottom of your coffee mug will go out. Should i try to add support for any type? Or are we ok with that? :D